WordPress/gutenberg
# AnglePickerControl: Use NumberControl as input field

## What?
Update the `AnglePickerControl` to use `NumberControl`

## Why?
Since the input field expects a number the `NumberControl` should be used. This adds a few enhancements that the `InputControl` doesn't have:

- It adds the ability to use the up/down arrows on the keyboard to change the number.
- It contrains the value to be between 0 and 360
- Input text is disallowed 
- Enabled the option to drag

## How?
Remove `InputControl` with `NumberControl`

## Testing Instructions
1. Test the component in storybook and make sure everything works as expected
2. Test the component where it is used and make sure everything works as expected. Ex. in the `GradientPicker`
3. Type in a too large number, ex. 9999. On blur the number should be contrained to 360.
4. Typing text should show in the input


## Screenshots or screencast 
| Before | After |
| - | - |
| ![before](https://user-images.githubusercontent.com/1415747/171281697-fd37b78c-eeef-491b-8b20-f5504d4d9846.gif) | ![after](https://user-images.githubusercontent.com/1415747/171281790-37f59850-df0a-41cf-851c-0734aed1f765.gif) |
